c7n/filters/health.py

```diff
@@ -21,43 +22,42 @@ class healthEventFilter(Filter):
 
     schema = type_schema(
             'health-events',
-            required=['eventTypeCodes'],
+            entityValues={'type': 'array', 'items': {'type': 'string'}},
             eventTypeCodes={'type': 'array', 'items': {'type': 'string'}},
             eventStatusCodes={'type': 'array', 'items': {'type': 'string'}})
     permissions = ('health:DescribeEvents', 'health:DescribeAffectedEntities',
         'health:DescribeEventDetails')
 
     def process(self, resources, event=None):
         results = []
-        if len(resources) > 0:
-            client = local_session(
-                self.manager.session_factory).client('health')
-            resource_map = self.get_resource_map(resources)
-            statusCodes = self.data.get(
-                'eventStatusCodes', ['open', 'upcoming'])
-            eventArns = []
-            paginator = client.get_paginator('describe_events')
-            for p in paginator.paginate(filter={
-                    'eventTypeCodes': self.data.get('eventTypeCodes'),
-                    'eventStatusCodes': statusCodes}):
-                eventArns.extend([e['arn'] for e in p['events']])
-            for arn in eventArns:
-                entity = client.describe_affected_entities(filter={
-                    'eventArns': [arn]})['entities'][0]['entityValue']
-                if entity in resource_map:
-                    eventDetail = client.describe_event_details(
-                        eventArns=[arn])
-                    resource_map[entity]['HealthEvent'] = {
-                        'entityValue': entity,
-                        'startTime': eventDetail['successfulSet'][0]
-                                        ['event']['startTime'],
-                        'eventTypeCode': eventDetail['successfulSet'][0]
-                                        ['event']['eventTypeCode'],
-                        'eventDescription': eventDetail['successfulSet'][0]
-                                        ['eventDescription']
-                                        ['latestDescription']}
-                    results.append(resource_map[entity])
+        if len(resources) == 0:
+            return results
+        client = local_session(self.manager.session_factory).client('health')
+        m = self.manager.get_model()
+        resource_map = {r[m.id]: r for r in resources}
+        paginator = client.get_paginator('describe_events')
+        statusCodes = self.data.get('eventStatusCodes', ['open', 'upcoming'])
+        f = {'services': [m.service.upper()], 'eventStatusCodes': statusCodes}
```

Comment:
might be worth reading the api docs on describe_events. boto3.readthedocs.io/en/latest/reference/services/health.html#Health.Client.describe_events